Framework/src/RootFileStorage.cxx

```diff
@@ -0,0 +1,370 @@
+    return {};
+  }
+  DirectoryNode currentNode{ pathToPos + 2, currentDir->GetName() };
+
+  TIter nextKey(currentDir->GetListOfKeys());
+  TKey* key;
+  while ((key = (TKey*)nextKey())) {
+    if (!loadObjects && std::strcmp(key->GetClassName(), MonitorObjectCollection::Class_Name()) == 0) {
+      std::string mocPath = currentNode.fullPath + std::filesystem::path::preferred_separator + key->GetName();
+      currentNode.children[key->GetName()] = MonitorObjectCollectionNode{ mocPath, key->GetName() };
+      continue;
+    }
+
+    ILOG(Debug, Devel) << "Getting the value for key '" << key->GetName() << "'" << ENDM;
+    auto* value = currentDir->Get(key->GetName());
+    if (value == nullptr) {
+      ILOG(Error) << "Could not get the value '" << key->GetName() << "', skipping." << ENDM;
+      continue;
+    }
+    if (auto moc = dynamic_cast<MonitorObjectCollection*>(value)) {
+      moc->postDeserialization();
+      std::string mocPath = currentNode.fullPath + std::filesystem::path::preferred_separator + key->GetName();
+      currentNode.children[moc->GetName()] = MonitorObjectCollectionNode{ mocPath, key->GetName(), moc };
+      ILOG(Info) << "Read object '" << moc->GetName() << "' in path '" << currentNode.fullPath << "'" << ENDM;
+    } else if (auto childDir = dynamic_cast<TDirectory*>(value)) {
+      currentNode.children[key->GetName()] = readStructureImpl(childDir, loadObjects);
+    } else {
+      ILOG(Warning, Devel) << "Could not cast the node to MonitorObjectCollection nor TDirectory, skipping." << ENDM;
+      delete value;
+      continue;
+    }
+  }
+
+  return currentNode;
+}
+
+MonitorObjectCollection* RootFileStorage::readMonitorObjectCollection(const std::string& path) const
+{
+  auto storedTObj = mFile->Get(path.c_str());
+  if (storedTObj == nullptr) {
+    ILOG(Error) << "Could not read file '" << path << "'" << ENDM;
+    return nullptr;
+  }
+  auto storedMOC = dynamic_cast<MonitorObjectCollection*>(storedTObj);
+  if (storedMOC == nullptr) {
+    ILOG(Error) << "Could not cast the stored object to MonitorObjectCollection" << ENDM;
+    delete storedTObj;
+  }
+  return storedMOC;
+}
+
+RootFileStorage::~RootFileStorage()
+{
+  if (mFile != nullptr) {
+    if (mFile->IsOpen()) {
+      ILOG(Info) << "Closing file '" << mFile->GetName() << "'." << ENDM;
+      mFile->Write();
+      mFile->Close();
+    }
+    delete mFile;
+  }
+}
+
+void deleteTDirectory(TDirectory* d)
+{
+  if (d != nullptr) {
+    d->Write();
+    d->Close();
+    delete d;
+  }
+}
+
+std::unique_ptr<TDirectory, void (*)(TDirectory*)> getOrCreateDirectory(TDirectory* parentDir, const char* dirName)
+{
+  auto dir = std::unique_ptr<TDirectory, void (*)(TDirectory*)>(parentDir->GetDirectory(dirName), deleteTDirectory);
+  if (dir == nullptr) {
+    ILOG(Info, Devel) << "Creating a new directory '" << dirName << "'." << ENDM;
```

Comment:
perhaps a debug message rather than an info ? I am not sure how often that is called though.